libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/DefaultAudioOffloadSupportProvider.java

```diff
@@ -81,14 +81,17 @@ public AudioOffloadSupport getAudioOffloadSupport(
 
     @C.Encoding
     int encoding = MimeTypes.getEncoding(checkNotNull(format.sampleMimeType), format.codecs);
-    if (encoding == C.ENCODING_INVALID) {
+    // AudioFormat.ENCODING_DTS_UHD_P2 is defined from API 34 onwards. We return offload
+    // unsupported to prevent crash in Util.getAudioFormat() below when it tries to create
+    // an AudioFormat with ENCODING_DTS_UHD_P2.
+    if ((encoding == C.ENCODING_INVALID) ||
+        ((Util.SDK_INT < 34) && (encoding == C.ENCODING_DTS_UHD_P2))) {
```

Comment:
I understand that that both conditions would lead to returning AudioOffloadSupport.DEFAULT_UNSUPPORTED. However, since this is a specific workaround with adjoining comment, could you split into two conditionals?

Reply:
@microkatz Done. Thanks.